api/src/main/java/io/minio/MinioClient.java

```diff
@@ -902,14 +902,7 @@ private Request createRequest(Method method, String bucketName, String objectNam
     }
 
     if (objectName != null) {
-      // Limitation: OkHttp does not allow to add '.' and '..' as path segment.
-      for (String pathSegment : objectName.split("/")) {
-        urlBuilder.addEncodedPathSegment(S3Escaper.encode(pathSegment));
-      }
-      if (objectName.endsWith("/")) {
-        // Fix issue #648: preserve trailing '/' to work with proxies like nginx.
-        urlBuilder.addPathSegment("");
-      }
+      urlBuilder.addEncodedPathSegments(S3Escaper.encodePath(objectName));
```

Comment:
You would need to retain comment about '.' and '..` where they are valid accordingly to S3 specification.

Reply:
What would be the purpose of that comment? @balamurugana MinIO server rejects path segments with `.` and `..` - do we still need to keep it?

Reply:
Because `.` and `..` are valid as per specification. As this library can be used with AWS S3 where it will fail and developer needs to know why it fails for such bug report.

Reply:
Done